features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/networksettled/ViewNetworkSettledMetric.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric.networksettled
+
+import com.datadog.android.api.InternalLogger
+import java.util.PriorityQueue
+import java.util.concurrent.TimeUnit
+
+internal class ViewNetworkSettledMetric(
+    requestIdentificationStrategy: NetworkSettledRequestIdentificationStrategy,
+    private val internalLogger: InternalLogger,
+    private val requestValidator: RequestValidator = resolveRequestValidator(requestIdentificationStrategy)
+) {
+
+    private val resourceStartedTimestamps = HashMap<String, HashMap<String, Long>>()
+    private val resourceStoppedTimestamps = HashMap<String, PriorityQueue<Long>>()
+    private val viewCreatedTimestampsByBViewId = HashMap<String, Long>()
```

Comment:
suggestion/ We need these maps because of using single instance of `VNSMetric` for the entire RUM application. If we instead create a distinct one for each RUM view scope there will be no need for using hash maps and likely the `purge()` routine will be gone.

Considering data flow and the domain of this problem (individual view metric), do we gain anything by sharing the same `VNSMetric` instance with all view scopes?

Reply:
you are totally right...I think there's nothing we can win here at least in this case. For the other metric we will have to share it so I am wondering if I should do it differently for this one ? What do you think ? It makes sense to remove the complexity though.

Reply:
One of the other reasons I wanted to share it was to not have to instantiate this all the time but I guess the memory penalty is not that high.